host/tests/dsDisplay_L3_Tests/dsDisplay_test02_GetDisplayEDIDBytes.py

```diff
@@ -0,0 +1,181 @@
+        """
+        Downloads the artifacts in test-setup configuration file to the dut.
+
+        Args:
+            None.
+        """
+
+        self.deviceDownloadPath = self.cpe.get("target_directory")
+
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+
+        # Download test artifacts to device
+        url = test.get("artifacts")
+        if url is not None:
+            self.downloadToDevice(url, self.deviceDownloadPath, self.rackDevice)
+
+    def testRunPrerequisites(self):
+        """
+        Runs Prerequisite commands listed in test-setup configuration file on the dut.
+
+        Args:
+            None.
+        """
+
+        #Run test specific commands
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+        cmds = test.get("execute")
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+    #TODO: Current version supports only manual verification.
+    def testGetEDIDBytes(self, manual=False, port="dsVIDEOPORT_TYPE_HDMI" , index:int = 0):
+        """
+        Gets the EDIDBytes of the display through the specified video port.
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                    Defaults to other verification methods
+                                    port (str, optional): The type of video port to retrieve the EDID Bytes.
+                                    index (int, optional): The index of the port to retrieve the EDID Bytes.
+        Returns:
+            bool
+        """
+        # Get the EDIDBytes from the display
+        edidbytes = self.testdsDisplay.getEdidBytes(port, index)
+
+        if manual:
+            # Manual verification
+            return self.testUserResponse.getUserYN(f"Is the EDID {edidbytes} correct? (Y/N):")
+        else:
+            # Automation verification
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """
+        This function will test the Display by getting the EDID Bytes of the display.
+
+        This function:
+        - Downloads the required assets.
+        - Runs the prerequisite commands.
+        - Retrieves EDID bytes for each supported port and verifies them.
+        - Cleans up assets after the test.
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        # Download the assets listed in the test setup configuration file
+        self.testDownloadAssets()
+
+        # Run Prerequisites listed in the test setup configuration file
+        self.testRunPrerequisites()
+
+        # Create an instance of the dsDisplayClass
+        self.testdsDisplay = dsDisplayClass(self.deviceProfile, self.hal_session)
+
+        # Initialize the dsDisplay module
+        self.testdsDisplay.initialise()
+
+        self.log.testStart(self.testName, '1')
```

Comment:
This number has to be the test case number, here 2 for test02

Reply:
Done